pkg/controlplane/control/manager.go

```diff
@@ -144,6 +144,99 @@ type Manager struct {
 	logger *logrus.Entry
 }
 
+// Restart coredns pods
+func coreDnsRestart(ctx context.Context, m *Manager) error {
+	var pods v1.PodList
+	if err := m.client.List(ctx, &pods, client.InNamespace("kube-system")); err != nil {
+		return err
+	}
+
+	for _, pod := range pods.Items {
+		if strings.Contains(pod.Name, "coredns") {
+			m.logger.Infof("Deleting pod: %s/%s.", pod.Namespace, pod.Name)
+			err := m.client.Delete(ctx, &pod)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// Add coredns rewrite for a given external dns service
```

Comment:
It may be less intrusive to use the `hosts` or `file` [plugins](https://coredns.io/plugins) so that the changes are not embedded as part of the CoreDNS configuration but are in their own file.
We could then ask an administrator to enable the CoreDNS plugin manually (if they so desire) and require lower privileges for us (we won't be mocking with the Corefile directly).

Reply:
@elevran , looking into `hosts` plugin, in case it is configured with malformed syntax, coredns pods seem to also fail.


```
$ kubectl --kubeconfig ~/.kube/cl-peer2  get pod   -n kube-system
NAME                                             READY   STATUS             RESTARTS      AGE
coredns-5d78c9869d-klk2t                         0/1     CrashLoopBackOff   5 (61s ago)   4m11s

$ kubectl --kubeconfig ~/.kube/cl-peer2  logs coredns-5d78c9869d-klk2t    -n kube-system
/etc/coredns/custom.hosts:1 - Error during parsing: Unknown directive '1.2.3.4'
```

**coredns configmap:**

```
apiVersion: v1
data:
  Corefile: |
    .:53 {
        import custom.hosts                    < ---- added
        errors
        health {
           lameduck 5s
        }
        ready
        kubernetes cluster.local in-addr.arpa ip6.arpa {
           pods insecure
           fallthrough in-addr.arpa ip6.arpa
           ttl 30
        }
        prometheus :9153
        forward . /etc/resolv.conf {
           max_concurrent 1000
        }
        cache 30
        loop
        reload
        loadbalance
        hosts custom.hosts                    < ---- added
    }
  custom.hosts: |                    < ---- added
    1.2.3.4 mycustom.host
kind: ConfigMap

```

**coredns deployment:**

```
...
      containers:
      - args:
        - -conf
        - /etc/coredns/Corefile
        image: registry.k8s.io/coredns/coredns:v1.10.1
        imagePullPolicy: IfNotPresent
...
        volumeMounts:
        - mountPath: /etc/coredns
          name: config-volume
          readOnly: true

...
      volumes:
      - configMap:
          defaultMode: 420
          items:
          - key: Corefile
            path: Corefile
          - key: custom.hosts                    < ---- added
            path: custom.hosts                    < ---- added
          name: coredns
        name: config-volume


```